### PR TITLE
GHW: adjust style, build shared lib, add to docs

### PR DESCRIPTION
**Pull request by umarcor**
The first two commits in this PR "fix" the license headers in subdir ghw, for making it consistent with other C sources, and then clang-format was used for adjusting the style.

In the third commit, shared library `ghwlib.[so|dll]` is built and installed with GHDL, by default.

In the last commit, a section about GHW is added to the documentation (in chapter "Development"). See https://umarcor.github.io/ghdl/ghw/index.html.

### CODE REVIEW

**Comment by tgingold** [comment]
I would prefer to use the GNU layout for C files.

**Comment by umarcor** [comment]
> I would prefer to use the GNU layout for C files.

Ok. I will search if there is any automatic formatter for that.

**Comment by umarcor** [comment]
Done. I used GNU indent, instead of clang-format.

File: Makefile.in [hunk 1]
```diff
@@ -479,19 +479,26 @@ ghwdump: ghwdump$(EXEEXT)
 ghwdump$(EXEEXT): $(GHWDUMP_OBJS)
 	$(CC) -o $@ $(GHWDUMP_OBJS)
 
+ghwlib$(SOEXT): ghwlib.o
```
Comment (Xiretza) [inline Makefile.in:482]: If this was named `libghw$(SOEXT)`, it could be linked against with `-lghw`.

**Comment by tgingold** [comment]
@Xiretza is right: a library name starts with lib.

For a follow-up.

File: Makefile.in (hunk 1 above)
Comment (umarcor) [reply Makefile.in:482]: Absolutely. And it's not the first time I/we do this exact same mistake... Thanks for catching it!

**Pull request merged**